### PR TITLE
finalize MP2020 correction values

### LINKED ISSUE #2056: latexify_ion is broken
**Describe the bug**
The `latexify_ion` method in `analysis.pourbaix_diagram` does not work; it just returns the input formula.

**To Reproduce**
```
from pymatgen.analysis.pourbaix_diagram import latexify_ion
latexify_ion('Na+')
'Na+'
latexify_ion('SO4-2')
'SO4-2'
```

**Expected behavior**
The formula would be reformatted with appropriate LateX sub/superscripts

**Desktop (please complete the following information):**
 - OS: Mac
 - Version: latest Github master as of Feb 3, 2021

### PR DESCRIPTION
**Pull request by rkingsbury**
## Summary

Final correction fitting for `MaterialsProject2020Compatibility`

* Updates the fitted values to be fully consistent with the new `expt_formation_energy_kingsbury` dataset (see https://github.com/hackingmaterials/matminer/pull/602)
* Use `MaterialsProject2020Compatibility` by default in `get_pourbaix_entries`
* Fixes `DeprecationWarning` in `pourbaix_diagram.py` related to `latexify_ion` (fixes #2056)

### CODE REVIEW

**Comment by rkingsbury** [comment]
@mkhorton there seem to be some problems with the test environment - it complains about enumlib and openbabel not being installed / available

**Comment by mkhorton** [comment]
That message is just saying those tests have been skipped -- the actual error is that PMG_MAPI_KEY isn't set. I can look into it.

File: pymatgen/ext/tests/test_matproj.py [hunk 1]
```diff
@@ -329,15 +329,15 @@ def test_get_pourbaix_entries(self):
...
 
         feo2 = [e for e in pbx_entries if e.entry_id == "mp-25332"][0]
-        self.assertAlmostEqual(feo2.energy, 2.5523680849999995, places=2)
+        self.assertAlmostEqual(feo2.energy, 3.56356, places=3)
 
         # Test S, which has Na in reference solids
         pbx_entries = self.rester.get_pourbaix_entries(["S"])
         so4_two_minus = pbx_entries[9]
-        self.assertAlmostEqual(so4_two_minus.energy, 0.0644980568750011, places=2)
+        self.assertAlmostEqual(so4_two_minus.energy, 0.301511, places=3)
```
Comment (mkhorton) [inline pymatgen/ext/tests/test_matproj.py:340]: @rkingsbury before merging, can you share verification that the new Pourbaix numbers are decent? have you compared the resulting Pourbaix diagrams? These differences seem like they may be significant.

Comment (rkingsbury) [reply pymatgen/ext/tests/test_matproj.py:340]: Hi @mkhorton thanks for flagging this; I should have said something about it. I saw it too and had the same reaction, but I checked many Pourbaix diagrams with the correction values in this MPR and they compare well with the existing ones. I'll email you a slide deck so you can see the comparisons. Bottom line I don't think we need to be concerned, and being able to do a full DB build will give us even more confidence.

**Comment by mkhorton** [comment]
Thanks @rkingsbury :)

**Pull request merged**